Migrate returns tests

Cruikshanks opened the pull request:
This change migrates the legacy returns tests to this project. The legacy tests were

- returns/invite-licence-holders-submit-returns.spec.js
- returns/remind-licence-holders-submit-returns.spec.js
- returns/returns-status-view.spec.js
- returns/update-view-returns-status-of-a-licence.spec.js

andrewhick commented on features/step_definitions/search_filter_steps.rb:
```diff
@@ -100,7 +101,12 @@
   else
-    expect(@front_app.licences_page).to have_licence_links_internal count: @expected_result_count.to_i
+    if @front_app.licences_page.has_licence_links_internal?
+      p @expected_result_count
```
puts

andrewhick commented on features/test:
```diff
@@ -0,0 +1,146 @@
+
```
Not sure why this email is here - can you remove it from the folder or add it to .gitignore, unless all data is fake?

andrewhick commented on oupput.html:
```diff
@@ -0,0 +1,476 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns ="http://www.w3.org/1999/xhtml"><head><meta http-equiv="Content-Type" content="text/html;charset=utf-8"/><title>Cucumber</title><style type="text/css">/* cucumber.css is generated from cucumber.sass */
```
Can you let me know the purpose of this file? Does it need adding to .gitignore?

andrewhick commented on features/page_objects/digitise_review_page.rb:
```diff
@@ -13,11 +13,14 @@ class DigitiseReviewPage < SitePrism::Page
   element(:no_radio, "#status-1", visible: false)
   element(:approved_radio, "#status-2", visible: false)
-  element(:lic_review_radio, "#status-3", visible: false)
+  element(:lic_review_radio, "#status-4", visible: false)
```
@swaff If this works now I'm happy with it as it is, but in future I think it would be better to identify based on inner text or something else to make the tests more resilient.

andrewhick commented on features/page_objects/front_office_app.rb:
```diff
@@ -193,6 +201,14 @@ def return_check_page
     @last_page = ReturnCheckPage.new
   end
 
+  def return_has_water_been_abstracted
```
If this is a simple radio button question, could it be absorbed in return_routes rather than having a separate page object file?

andrewhick commented on features/page_objects/return_routes_page.rb:
```diff
@@ -1,33 +1,25 @@
   # These use the same identifiers but different options.
-  element(:enter_radio, "input[id='radio-inline-0']", visible: false)
-  element(:query_radio, "input[id='radio-inline-1']", visible: false)
+  element(:enter_radio, ".govuk-radios__item:nth-child(1) .govuk-radios__input", visible: false)
```
Is there any way to avoid nth-child here? If it's the only option then fine, but I feel these can make tests flaky if they move around.

andrewhick commented on features/step_definitions/agent_steps.rb:
```diff
@@ -44,12 +40,10 @@
-    confirmpw: Quke::Quke.config.custom["data"]["accounts"]["password"]
-  )
+  password = config_accounts("password")
+  @front_app.register_create_pw_page.submit_password password
```
This looks like a change to existing syntax. Is it possible to make it consistent with the rest of the pack along the lines of:
```
  @front_app.register_create_pw_page.submit(
    email: email,
    password: password
  )
```
...or alternatively, bring the rest of the pack in line with the new formatting?

andrewhick commented on features/support/helpers.rb:
```diff
@@ -11,13 +11,58 @@ def scroll_to(element)
+# - external water resources
+# - internal admin
+# - developer admin
+def config_urls
```
I love this as a way to simplify the syntax

Cruikshanks committed:
Migrate returns tests

This change migrates the legacy returns tests to this project. The legacy tests were

- returns/invite-licence-holders-submit-returns.spec.js
- returns/remind-licence-holders-submit-returns.spec.js
- returns/returns-status-view.spec.js
- returns/update-view-returns-status-of-a-licence.spec.js

Cruikshanks committed:
Merge branch 'main' into migrate-returns-tests
